### PR TITLE
fix: title of journey from template

### PR DESCRIPTION
**Pull request by storyworks**
# Description

We get "copy 0" whenever we use a template from the template library. 
We want it just to display the template name, or add copy only if another journey with the template name already exists in our journey list.
image

[- Link to Basecamp Todo
](https://3.basecamp.com/3105655/buckets/31822575/todos/5909445092)

# How should this PR be QA Tested?

Please describe the QA tests that you ran to verify your changes. Please also note any relevant details for your test configuration.

- [x] Go to the template library and use a template to create a journey. The created journey should have the same name as the template.
- [x] Repeat the step above, this time the created journey should have the template name + "copy" at the end.

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/api-journeys/src/app/modules/journey/journey.resolver.ts [hunk 1]
```diff
@@ -284,8 +284,12 @@ export class JourneyResolver {
       journey.title
     )
     const duplicateNumber = this.getFirstMissingNumber(duplicates)
-    const duplicateTitle = `${journey.title} copy ${
-      duplicateNumber === 1 ? '' : duplicateNumber
+    const duplicateTitle = `${journey.title}${
```
Comment (csiyang) [inline apps/api-journeys/src/app/modules/journey/journey.resolver.ts:287]: Not the biggest fan of chaining ternaries like this, makes it kinda hard to read. Thoughts on splitting the logic out abit?

Comment (storyworks) [reply apps/api-journeys/src/app/modules/journey/journey.resolver.ts:287]: Personally I think 2 levels formatted like this is as readable as a switch statement. Eg first line is the case, next line the result, then case, then result, last line is default.

What would you change it to?

Comment (csiyang) [reply apps/api-journeys/src/app/modules/journey/journey.resolver.ts:287]: Like just splitting out one of the ternaries

```
  const x = duplicateNumber > 1 ? duplicateNumber : ''
  const duplicateTitle = duplicateNumber === 0 ? '' : ` copy ${x}`
```

Comment (storyworks) [reply apps/api-journeys/src/app/modules/journey/journey.resolver.ts:287]: Your one you've added an extra " " when duplicate number is 1. Even if you rewrite it to:

` const x = duplicateNumber > 1 ? ` ${duplicateNumber}` : ''`
I think that's still more complex that what I have :/

Comment (csiyang) [reply apps/api-journeys/src/app/modules/journey/journey.resolver.ts:287]: Your trim should handle any trailing spaces right? Its easier to read and understand, but its just what I would do.

Comment (storyworks) [reply apps/api-journeys/src/app/modules/journey/journey.resolver.ts:287]: Agree to disagree?

Comment (csiyang) [reply apps/api-journeys/src/app/modules/journey/journey.resolver.ts:287]: sure lol

**Review by csiyang — APPROVED** [review]

File: apps/api-journeys/src/app/modules/journey/journey.resolver.ts (hunk 1 above)
Comment (storyworks) [reply apps/api-journeys/src/app/modules/journey/journey.resolver.ts:287]: Also I take back, I have nested the ternaries but only 1 level. We haven't enabled `/*eslint no-nested-ternary: "error"*/` so I've not violated our own linting standards. 

If we want to impose this rule, then it's quite easy to do but has not been done.
I do get where people are going when they say don't nest excessively and would agree that 2 levels or 1 level but super complex logic would make it difficult to read. But I think 1 level especially for something so simple is better than splitting up logic just for the sake of never nesting ternaries.

**Pull request merged**